client/state/videos/actions.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	VIDEO_FETCH,
+	VIDEO_FETCH_COMPLETED,
+	VIDEO_FETCH_FAILED,
+	VIDEO_RECEIVE
+} from 'state/action-types';
+
+/**
+ * Triggers a network request to fetch a video.
+ *
+ * @param  {String}    guid VideoPress guid
+ * @return {Function}       a promise that will resolve once fetching is completed
+ */
+export function fetchVideo( guid ) {
+	return ( dispatch ) => {
+		dispatch( {
+			type: VIDEO_FETCH,
+			guid
+		} );
+
+		return new Promise( ( resolve ) => {
+			wpcom.undocumented().videos( guid, ( error, data ) => {
```

Comment:
If you change your undocumented `wpcom` to return the result of `this.wpcom.req.get`, you can return this call as a promise, avoiding the need for a wrapper at all.